connectors/src/connectors/slack/chat/citations.ts

```diff
@@ -18,11 +19,36 @@ export function annotateCitations(
   content: string,
   action: AgentActionType | null
 ): { formattedContent: string; footnotes: SlackMessageFootnotes } {
-  const references: { [key: string]: RetrievalDocumentType } = {};
+  const references: {
```

Comment:
Why not keep the raw types (RetrievalDocumentType | WebSearchSomething) here as we might want to do more with citations in the future at the display level?

Reply:
That's what I had just before and I got a feedback to simplify as is for the second loop 🙇🏻‍♀️